lua/opencode/config.lua

```diff
@@ -26,51 +28,53 @@ M.defaults = {
-      diff_close = '<leader>oc',
-      diff_revert_all_last_prompt = '<leader>ora',
-      diff_revert_this_last_prompt = '<leader>ort',
-      diff_revert_all = '<leader>orA',
-      diff_revert_this = '<leader>orT',
-      diff_restore_snapshot_file = '<leader>orr',
-      diff_restore_snapshot_all = '<leader>orR',
-      swap_position = '<leader>ox', -- Swap Opencode pane left/right
-      permission_accept = '<leader>opa',
-      permission_accept_all = '<leader>opA',
-      permission_deny = '<leader>opd',
+      ['<leader>og'] = 'toggle',
+      ['<leader>oi'] = 'open_input',
+      ['<leader>oI'] = 'open_input_new_session',
+      ['<leader>oo'] = 'open_output',
+      ['<leader>ot'] = 'toggle_focus',
+      ['<leader>oq'] = 'close',
+      ['<leader>os'] = 'select_session',
+      ['<leader>op'] = 'configure_provider',
+      ['<leader>od'] = 'diff_open',
+      ['<leader>o]'] = 'diff_next',
+      ['<leader>o['] = 'diff_prev',
+      ['<leader>oc'] = 'diff_close',
+      ['<leader>ora'] = 'diff_revert_all_last_prompt',
+      ['<leader>ort'] = 'diff_revert_this_last_prompt',
+      ['<leader>orA'] = 'diff_revert_all',
+      ['<leader>orT'] = 'diff_revert_this',
+      ['<leader>orr'] = 'diff_restore_snapshot_file',
+      ['<leader>orR'] = 'diff_restore_snapshot_all',
+      ['<leader>oC'] = 'open_configuration_file',
+      ['<leader>ox'] = 'swap_position',
+      ['<leader>opa'] = 'permission_accept',
+      ['<leader>opA'] = 'permission_accept_all',
+      ['<leader>opd'] = 'permission_deny',
     },
     window = {
-      submit = '<cr>',
-      submit_insert = '<cr>',
-      close = '<esc>',
-      stop = '<C-c>',
-      next_message = ']]',
-      prev_message = '[[',
-      mention_file = '~',
-      mention = '@',
-      slash_commands = '/',
-      toggle_pane = '<tab>',
-      prev_prompt_history = '<up>',
-      next_prompt_history = '<down>',
-      switch_mode = '<M-m>',
-      focus_input = '<C-i>',
-      select_child_session = '<leader>oS',
-      debug_message = '<leader>oD',
-      debug_output = '<leader>oO',
-      debug_session = '<leader>ods',
-      permission_accept = 'a',
-      permission_accept_all = 'A',
-      permission_deny = 'd',
+      ['<cr>'] = { 'submit_input_prompt', mode = { 'n', 'i' } },
+      ['<esc>'] = 'close',
+      ['<C-c>'] = 'stop',
+      [']]'] = 'next_message',
+      ['[['] = 'prev_message',
+      ['~'] = { 'mention_file', mode = 'i' },
+      ['@'] = { 'mention', mode = 'i' },
+      ['/'] = { 'slash_commands', mode = 'i' },
+      ['<tab>'] = { 'toggle_pane', mode = { 'n', 'i' } },
+      ['<up>'] = { 'prev_prompt_history', mode = { 'n', 'i' } },
+      ['<down>'] = { 'next_prompt_history', mode = { 'n', 'i' } },
+      ['<M-m>'] = 'switch_mode',
+      ['<C-i>'] = 'focus_input',
+      ['<leader>oS'] = 'select_child_session',
+      ['<leader>oD'] = 'debug_message',
+      ['<leader>oO'] = 'debug_output',
+      ['<leader>ods'] = 'debug_session',
+    },
```

Comment:
I think we should separate input and output window keymaps.

```lua
keymaps = {
    editor = {
        
    },
    input_window = {
           ['<down>'] = { 'next_prompt_history', mode = { 'n', 'i' } },
           ...
    },
    output_window = {
          ['<C-i>'] = {'focus_input'}, -- only 1 format to simplify ?
           ...
    },
   global = {
        ...legacy
    },
    window = {
        -- legacy
    }
}

```

This should make the legacy detection easier at the expanse of duplicating some keymaps for flexibility